### DISCUSSION TITLE
Get Total Tags count under root folder and Read unit/description of a Node

### DISCUSSION THREAD

**Discussion by yasodakrishnav** [eclipse-milo/milo#1145] [General]
Hey Team,

1. Is there any that i can get the total number tags count under root folder using milo in java?
2. How can i read the unit and description of the tag while browsing the ReferenceDescription?

Thanks.

**Comment by kevinherron** [comment]
> Is there any that i can get the total number tags count under root folder using milo in java?

The only way to accomplish such a thing would be a full recursive browse, which is probably not advisable. This isn’t a Milo limitation so much as it is an OPC UA limitation.

> How can i read the unit and description of the tag while browsing the ReferenceDescription?

These would have to be read in a separate step. The [ReferenceDescription](https://github.com/eclipse/milo/blob/master/opc-ua-stack/stack-core/src/main/java/org/eclipse/milo/opcua/stack/core/types/structured/ReferenceDescription.java) structure returned by the browse service is defined by the OPC UA spec.

**Reply by yasodakrishnav** [reply]
Could you please give me an example to how to read the unit and description?

Thanks for the immediate response kevin.

**Reply by kevinherron** [reply]
I’m away on vacation and only have an iPad so I can’t really give you any code examples until next week.

Description is a built-in OPC UA attribute you would read using the Read service. I assume by unit you mean Engineering Unit, which is a Property that only Nodes of type AnalogItemType have. You would have to look at the type definition of each reference to see if it’s an AnalogItemType or subclass, or as you browse recursively and find a PropertyNode with BrowseName “EngineeringUnits” assume that it is, and then read the Value attribute of that Property Node.

**Reply by yasodakrishnav** [reply]
Alright Kevin. Thanks for your response and time.  Would be great if you provide me the code when you are back.
